### ISSUE TITLE
add webdav support

### ISSUE THREAD

**Issue by gtsteffaniak** [gtsteffaniak/filebrowser#209]
as mentioned here https://github.com/filebrowser/filebrowser/issues/3379

**Comment by leopoldhub** [comment]
Hello, I too would be very interested by this feature.

**Comment by colin715** [comment]
Would really appreciate this feature. Love the software so far!

**Comment by reddec** [comment]
I am considering to start working on it this XMas. If I will become MIA 🙂  - here there reference implementation that I am about to be inspired by

- https://github.com/mholt/caddy-webdav/blob/master/webdav.go
- https://pkg.go.dev/golang.org/x/net/webdav

fyi @gtsteffaniak

**Comment by gtsteffaniak** [comment]
Thanks keep me updated here. I haven't looked into any WebDAV capability yet.

**Comment by reddec** [comment]
@gtsteffaniak I added initial support, however, may I ask your advise about API keys?
The current implementation uses JWT which usually longer than many clients can support as basic auth (see PR notes).

**Comment by gtsteffaniak** [comment]
Yep. JWT tokens will remain the same for filebrowser side. I don't want to reduce the security of filebrowser to accommodate WebDAV's security limitations.

I will respond on PR

**Comment by piyushaswani55** [comment]
I have tried mounting my source as webdav on my Macbook Finder as well as MaterialFiles & Cx File Explorer on Android with `1.2.x` releases, and it has worked just as expected for me.
This feature enhancement has made me love Filebrowser Quantum even more.
Thank you @gtsteffaniak and @reddec 🥳

**Comment by gtsteffaniak** [comment]
just an fyi, this is techinically supported in v1.2.0 but it doesn't respect access rules or indexing settings. Official support will come in v1.3.0!

**Comment by gtsteffaniak** [comment]
its officially live on v1.3.0 -- and on the [documentation](https://filebrowserquantum.com/en/docs/features/webdav/)!

**Comment by gtsteffaniak** [comment]
gonna mark this closed.
